CDCgov/prime-reportstream
# Victor/8096 remove FHIR resources

This PR adds a couple of utility functions under `FHIRBundleHelpers` to be able to remove FHIR resources from a bundle. The function will be used specifically to remove Observations for now, but it can be used to remove any resource. The `deleteResource` function will remove the resource and any orphaned resources in the bundle.

Test Steps:
1. This functionality can only be tested using unit tests for now. 
2. Run the unit tests inside `FHIRBundleHelpersTests` and make sure the tests pass
3. Optional: Inspect the FHIR bundle after the resources have been deleted and spot check the resources have been deleted correctly and their references as well.

## Changes
- *Added utility functions  to safely delete a FHIR bundle resource

## Checklist

### Testing

## Linked Issues
- Fixes #8096
